packages/docs/docs/react/use-subscription.md

```diff
@@ -0,0 +1,235 @@
+---
+sidebar_position: 3
+tags:
+  - subscription
+---
+
+# `useSubscription` Hook
+
+`useSubscription` creates an in-memory `Subscription` resource with the given criteria on the Medplum server and calls the 
+given callback when an event notification is triggered by a resource interaction over a WebSocket connection.
+
+Subscriptions created with this hook are lightweight, share a single WebSocket connection per instance of `MedplumClient`, and are automatically 
+untracked and cleaned up when the containing component is no longer mounted.
+
+## Use Cases
+
+The `useSubscription` hook is a powerful tool for creating applications that require the client to wait for events asynchronously from the server.
+
+Some examples include:
+- Live chat
+- Notification badges for tasks
+- Realtime analytics dashboards
+
+The semantics of the hook make it extremely simple to subscribe to events for resource interactions that satisfy a specified criteria and act
+on them reactively, rather than having to poll via expensive search requests on a timer.
+
+## Usage
+
+The `useSubscription` hook takes a [FHIR search criteria](https://hl7.org/fhir/R4/search.html) and a callback function to call when a resource interaction
+that satisfies the Subscription occurs.
+
+```tsx
+function MyComponent(): JSX.Element {
+  const [notificationCount, setNotificationCount] = useState(0);
+
+  useSubscription(
+    'Communication?sender=Practitioner/abc-123&recipient=Practitioner/me-456', 
+    (bundle: Bundle) => {
+      console.log('Received a message from Practitioner/abc-123!');
+      handleNotificationBundle(bundle); // Do something with the bundle
+      setNotificationCount(s => s + 1);
+    }
+  );
+
+  return <div>Notifications received: {notificationCount}</div>;
+}
+```
+
+### Parsing the Subscription `Bundle`
+
+The callback receives a `Bundle` which contains a `SubscriptionStatus` as its first entry (`bundle.entry[0].resource`) 
+and the resource for which the `Subscription` fired for in its second entry (`bundle.entry[1].resource`).
```

Comment:
It might be more useful to give a full example of the JSON received by the callback, showing what both Bundle entries actually look like